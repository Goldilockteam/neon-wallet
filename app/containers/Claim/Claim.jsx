// @flow
import React, { Component } from 'react'

import Button from '../../components/Button'
import Tooltip from '../../components/Tooltip'
import { formatGAS } from '../../core/formatters'
import { toBigNumber } from '../../core/math'
import ClaimIcon from '../../assets/icons/claim.svg'

type Props = {
  className: ?string,
  doGasClaim: Function,
  disableClaimButton: boolean,
  claimAmount: string
}

export default class Claim extends Component<Props> {
  intervalId: ?number

  render() {
    const { className, claimAmount } = this.props
    const disabled = this.isDisabled()

    return (
      <div>
        <Tooltip
          title={
            toBigNumber(claimAmount).eq(0)
              ? 'Address must hold NEO in order to claim GAS'
              : 'You can claim GAS once every 5 minutes'
          }
          disabled={!disabled}
        >
          <Button
            id="claim"
            className={className}
            disabled={disabled}
            primary
            renderIcon={ClaimIcon}
            onClick={this.handleClaim}
          >
            Claim {this.getFormattedAmount()} GAS
          </Button>
        </Tooltip>
      </div>
    )
  }

  handleClaim = () => {
    this.props.doGasClaim()
  }

  isDisabled = () => {
    const { claimAmount, disableClaimButton } = this.props
    return disableClaimButton || toBigNumber(claimAmount).eq(0)
  }

  getFormattedAmount = () => formatGAS(this.props.claimAmount)
}
