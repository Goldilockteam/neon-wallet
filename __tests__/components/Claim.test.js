import React from 'react'
import { shallow, mount } from 'enzyme'

import Claim from '../../app/containers/Claim/Claim'

describe('Claim', () => {
  const props = {
    doGasClaim: () => {},
    disableClaimButton: false,
    claimAmount: '1.25406935'
  }

  test('should render claim GAS button as enabled', () => {
    const wrapper = shallow(<Claim {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('should render claim GAS button as disabled', () => {
    const wrapper = shallow(<Claim {...props} disableClaimButton />)
    expect(wrapper).toMatchSnapshot()
  })

  test('should claim GAS when button is clicked', () => {
    const claimSpy = jest.fn()
    const wrapper = mount(<Claim {...props} doGasClaim={claimSpy} />)

    wrapper.find('button#claim').simulate('click')

    expect(claimSpy).toHaveBeenCalled()
  })
})
