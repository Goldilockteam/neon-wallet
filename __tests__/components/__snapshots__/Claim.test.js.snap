// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Claim should render claim GAS button as disabled 1`] = `
<div>
  <Tooltip
    disabled={false}
    title="You can claim GAS once every 5 minutes"
  >
    <Button
      cancel={false}
      disabled={true}
      id="claim"
      onClick={[Function]}
      primary={true}
      secondary={false}
    >
      Claim 
      1.25406935
       GAS
    </Button>
  </Tooltip>
</div>
`;

exports[`Claim should render claim GAS button as enabled 1`] = `
<div>
  <Tooltip
    disabled={true}
    title="You can claim GAS once every 5 minutes"
  >
    <Button
      cancel={false}
      disabled={false}
      id="claim"
      onClick={[Function]}
      primary={true}
      secondary={false}
    >
      Claim 
      1.25406935
       GAS
    </Button>
  </Tooltip>
</div>
`;
